### PR TITLE
Issue548 gh pages in master

### PR DESCRIPTION
**Pull request by jsmailes**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Changes the process of building and deploying documentation so `make build-gh-pages` is no longer necessary.

The contents of the `gh-pages` branch has been moved to the `docs` folder in the `master` branch.

On push to the `master` branch (or when a pull request is merged with `master`), a GitHub Action is triggered which pushes any changes to the `docs` folder to the `gh-pages` branch.

This does not change behaviour of the site and does not require any configuration changes, but it improves the repository's usability by getting rid of the need to keep track of two long-live branches.

Does this close any currently open issues?
------------------------------------------
This closes issues #484 and #548 by unifying the `gh-pages` and `master` branches and automating updates of the site based on changes to the `master` branch.

Any relevant logs, error output, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

Any other comments?
-------------------
It may be possible to make this change using Travis instead of GitHub Actions if desired.

I am more familiar with GitHub Actions so that's what I've used here, but I imagine it would be possible to get the same results with a Travis configuration file.

### CODE REVIEW

**Comment by joshrabinowitz** [comment]
@jsmailes thank you for this PR! I really appreciate it.

Can you please give detailed instructions how I should test this branch?

I'm sort of drawing a blank -- I don't typically use git hooks or github actions -- but maybe I should!

File: .github/workflows/github-pages.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+          submodules: recursive
```
Comment (sobolevn) [inline .github/workflows/github-pages.yml:15]: We don't have any submodules. So, this can be skipped (I guess).

**Review by sobolevn — APPROVED** [review]
Great work! 👍

File: .github/workflows/github-pages.yml (hunk 1 above)
Comment (joshrabinowitz) [reply .github/workflows/github-pages.yml:15]: @sobolevn did you test this branch?

Comment (sobolevn) [reply .github/workflows/github-pages.yml:15]: Not yet.

**Comment by joshrabinowitz** [comment]
Update for @jsmailes - I'll look at this in depth after we get the next release out.

**Comment by joshrabinowitz** [comment]
OK, I've looked this over. 

The ideal solution, I think, would be to stop using the `gh-pages` branch all together, and instead put all the files for `git-secret.io` in a directory somewhere like `.gh-pages/`, in the _master_ branch.

Is this doable?

**Comment by sobolevn** [comment]
Yes, github can build docs from `docs/` folder now.

**Comment by joshrabinowitz** [comment]
I think we should do what @sobolevn describes above, and use the (magical) `/docs` folder in `master` instead of the magical `gh-pages` branch

@jsmailes, would you be amenable to re-doing this PR using just the /docs folder and without depending on the `gh-pages` bransch?

**Review by sobolevn — APPROVED** [review]

**Pull request merged**